docs/istio.md

```diff
@@ -0,0 +1,71 @@
+3. **Cleanup:** Once the new version is verified, a follow-up PR removes the old revision from `svc.istio.versions`. The next pipeline run removes the obsolete revision from the cluster.
+
+Before any revision alias is moved or workloads are restarted, `istio.sh` verifies the health and readiness of the target mesh defined by `svc.istio.targetVersion`. This ensures a safe transition and reduces the risk of service disruption during upgrades. If issues arise during an upgrade, the `svc.istio.targetVersion` can be reverted. This causes the revision tag to point back to the previous revision, allowing workloads to seamlessly revert to the old mesh while retaining the new version on-cluster for analysis.
+
+This upgrade mechanism is tightly integrated with the cluster management pipeline [svc-pipeline.yaml](../dev-infrastructure/svc-pipeline.yaml). All operations, including tagging and workload restarts, are idempotent and safe to re-run, ensuring consistent and predictable mesh transitions.
+
+## Workload Mesh Integration
+
+Workload integration with the Istio mesh is achieved through namespace-level opt-in via revision-based labeling. Each service component's `Makefile` is responsible for applying the `istio.io/rev` label to its deployment namespace. This label points to the currently active Istio revision alias (e.g., `prod-stable`), which is sourced from the configuration and passed into the Makefile environment.
+
+Istio sidecar injection is enabled automatically for all pods within a labeled namespace. No explicit pod-level annotations are required. This approach ensures consistent and centralized control over mesh participation.
+
+## Traffic Control and Security
+
+Istio is used to enforce strict service-to-service communication policies within the service cluster. Each service component is responsible for defining their own `AuthorizationPolicy` resources as part of its Helm chart, adhering to the following principles:
+
+* **Default deny:** Every namespace must include a `deny all` policy to ensure that no traffic is permitted unless explicitly allowed.
+* **Per-workload allow policies:** Workloads that need to receive traffic must define one or more `AuthorizationPolicy` rules. These are scoped using selectors based on workload labels and may restrict traffic by port or even specific HTTP paths (e.g., `/metrics` for Prometheus scraping).
+
+Traffic origin is validated using `source.principal` fields where applicable, restricting access based on the SPIFFE identity tied to the source workload’s namespace and service account. This provides a strong and auditable trust boundary between services.
+
+## Ingress Management
+
+Ingress traffic to the service cluster is handled through a single shared Istio ingress gateway. This gateway is configured to terminate TLS traffic and route external requests to the RP frontend.
```

Comment:
Do you think it is relevant to mention that that istio Ingress Gateway is configured to use a dedicated PIP (tagged)?